imap_processing/utils.py

```diff
@@ -319,6 +319,11 @@ def packet_file_to_datasets(
     packet definition, the ``derived_value`` will be the converted value.
     The dimension of the dataset will be the time field in J2000 nanoseconds.
 
+    Note: While this is a general function intended to work for all
+    instruments, this cannot be used for IDEX due to the complex nature of
+    the xml file. IDEX will continue to use PacketParser and RawDustEvents
+    found in idex_l1.py.
```

Comment:
Sweet! Yeah, I agree with you. I changed the comment to a "Notes" section in the doc-string. 

I didn't end up implementing the dimension check in this PR (I just don't quite understand the code yet, and I feel like I should before I add it and test it). Would you like me to write another issue ticket?

Reply:
Yeah, that'd be great. I think the code above is close to working, so if you copy that over to another issue and work from that it might be a good thing. Additionally, add a test trying to parse IDEX and showing that your `ValueError` is raised with `pytest.raises(ValueError, match="Cannot prase apid")`.